js/lib/arrow.ts

```diff
@@ -81,27 +176,71 @@ function _loadFooter(bb) {
 
     var footerLengthOffset: number = fileLength - MAGIC.length - 4;
     bb.setPosition(footerLengthOffset);
-    var footerLength: number = Int64FromByteBuffer(bb, footerLengthOffset)
+    var footerLength: number = Int32FromByteBuffer(bb, footerLengthOffset)
 
     if (footerLength <= 0 || footerLength + MAGIC.length*2 + 4 > fileLength)  {
       console.log("Invalid footer length: " + footerLength)
     }
 
     var footerOffset: number = footerLengthOffset - footerLength;
     bb.setPosition(footerOffset);
-    var footer = arrow.flatbuf.Footer.getRootAsFooter(bb);
+    var footer = Footer.getRootAsFooter(bb);
 
     return footer;
 }
 
-function Int64FromByteBuffer(bb, offset) {
+function _loadSchema(bb) {
+    return _loadMessage(bb, MessageHeader.Schema, new Schema()).header;
+}
+
+function _loadRecordBatch(bb) {
+    var i, loaded = _loadMessage(bb, MessageHeader.RecordBatch, new RecordBatch());
+    if (loaded == null) {
+        return;
+    }
+    var nodes_ = [], nodesLength = loaded.header.nodesLength();
+    var buffer, buffers_ = [], buffersLength = loaded.header.buffersLength();
+
+    for (i = 0; i < nodesLength; i += 1) {
+        nodes_.push(loaded.header.nodes(i));
+    }
+    for (i = 0; i < buffersLength; i += 1) {
+        buffer = loaded.header.buffers(i);
+        buffers_.push({ offset: bb.position() + buffer.offset().low, length: buffer.length().low });
+    }
+    // position the buffer after the body to read the next message
+    bb.setPosition(bb.position() + loaded.length);
+
+    return { nodes: nodes_, buffers: buffers_, length: loaded.header.length().low };
+}
```

Comment:
I wonder if it will be more efficient to put the batch loading inside an "ArrayComponentSource" a la https://github.com/apache/arrow/blob/master/cpp/src/arrow/loader.h#L75, so that the Flatbuffer can be processed as needed while loading the record batch. Can do this later